spec/datadog/core/utils/network_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'spec_helper'
+
+require 'datadog/core/utils/network'
+
+RSpec.describe Datadog::Core::Utils::Network do
+  describe '.raw_ip_from_request_headers' do
+    context 'with default IP headers' do
+      context 'iterates over the deafult headers (DEFAULT_IP_HEADERS_NAMES) in order' do
+        it 'returns the first valid public IP value' do
+          headers = Datadog::Core::HeaderCollection.from_hash(
+            { 'X-Forwarded-For' => '10.42.42.42',
+              'True-Client-Ip' => '43.43.43.43',
+              'X-Cluster-Client-Ip' => '10.0.0.1', }
+          )
+
+          result = described_class.raw_ip_from_request_headers(headers)
+          expect(result).to eq('43.43.43.43')
+        end
+      end
+
+      context 'multiple IP addresses present in the header' do
+        it 'returns the first valid public IP address' do
+          headers = Datadog::Core::HeaderCollection.from_hash({ 'X-Forwarded-For' => '10.42.42.42,43.43.43.43,fe80::1' })
+
+          result = described_class.raw_ip_from_request_headers(headers)
+          expect(result).to eq('43.43.43.43')
+        end
+      end
+
+      context 'with custom header value' do
+        it 'returns the IP value if valid public address' do
+          headers = Datadog::Core::HeaderCollection.from_hash(
+            {
+              'X-Forwarded-For' => '64.233.161.147',
+              'test-header' => '43.43.43.43',
+            }
+          )
+
+          result = described_class.raw_ip_from_request_headers(headers, ip_headers_to_check: ['test-header'])
+          expect(result).to eq('43.43.43.43')
+        end
+
+        it 'returns nil if header not present' do
+          headers = Datadog::Core::HeaderCollection.from_hash({})
+
+          result = described_class.raw_ip_from_request_headers(headers, ip_headers_to_check: ['test-header'])
+          expect(result).to be_nil
+        end
+
+        it 'returns nil if header value is not valid' do
+          headers = Datadog::Core::HeaderCollection.from_hash({ 'test-header' => 'dd' })
+
+          result = described_class.raw_ip_from_request_headers(headers, ip_headers_to_check: ['test-header'])
+          expect(result).to be_nil
+        end
+      end
+
+      it 'returns nil if no public valid IP addresss present in the headers' do
+        headers = Datadog::Core::HeaderCollection.from_hash(
+          { 'X-Forwarded-For' => '10.42.42.42' }
+        )
+
+        result = described_class.raw_ip_from_request_headers(headers)
+        expect(result).to be_nil
+      end
+    end
+  end
+
+  describe '.parse_ip' do
+    context 'valid IP' do
+      it 'returns the IP value, with port and zone idenifier removed' do
```

Comment:
Minor: Noticed a typo! :)

```suggestion
      it 'returns the IP value, with port and zone identifier removed' do
```